tests/unit/test_scheduler.py

```diff
@@ -109,3 +112,58 @@ def test_release_queued_tasks__auto_restart():
     # preparing ones
     mock_schd.pool.release_queued_tasks.assert_not_called()
     mock_schd.task_job_mgr.submit_task_jobs.assert_called()
+
+
+def test_auto_restart_DNS_error(monkeypatch, caplog, log_filter):
+    """Ensure that DNS errors in host selection are caught."""
+    def _select_workflow_host(cached=False):
+        # fake a "get address info" error
+        # this error can occur due to an unknown host resulting from broken
+        # DNS or an invalid host name in the global config
+        raise socket.gaierror('elephant')
+
+    monkeypatch.setattr(
+        'cylc.flow.scheduler.select_workflow_host',
+        _select_workflow_host,
+    )
+    schd = Mock(
+        workflow='myworkflow',
+        options=Mock(abort_if_any_task_fails=False),
```

Comment:
Don't need to mock this
```suggestion
        options=RunOptions(abort_if_any_task_fails=False),
```